src/views/DetailView.js

```diff
@@ -11,21 +11,51 @@ export const DETAIL_VIEW_ID = 'detail';
  * */
 export default class DetailView extends VivView {
   getLayers({ props, viewStates }) {
-    const { loader } = props;
+    const {
+      loader,
+      loaderSelection,
+      newLoaderSelection,
+      onViewportLoad,
+      transitionFields
+    } = props;
     const { id, height, width } = this;
     const layerViewState = viewStates[id];
     const layers = [];
 
-    const detailLayer = loader.isPyramid
-      ? new MultiscaleImageLayer(props, {
+    // Create at least one layer even without loaderSelection so that the tests pass.
+    if (loader.isPyramid) {
+      layers.push(
+        ...[loaderSelection, newLoaderSelection]
+          .filter((s, i) => i === 0 || s)
+          .map((s, i) => {
+            const suffix = s
+              ? `-${transitionFields.map(f => s[0][f]).join('-')}`
+              : '';
+            const newProps =
+              i !== 0
+                ? {
+                    onViewportLoad,
+                    refinementStrategy: 'never',
+                    excludeBackground: true
+                  }
+                : {};
+            return new MultiscaleImageLayer({
+              ...props,
+              ...newProps,
+              loaderSelection: s,
+              id: `${loader.type}${getVivId(id)}${suffix}`,
+              viewportId: id
+            });
+          })
+      );
+    } else {
+      layers.push(
+        new ImageLayer(props, {
           id: `${loader.type}${getVivId(id)}`,
           viewportId: id
         })
-      : new ImageLayer(props, {
-          id: `${loader.type}${getVivId(id)}`,
-          viewportId: id
-        });
-    layers.push(detailLayer);
+      );
+    }
```

Comment:
I didn't try. If you have a sample image that has a z-stack but no pyramid, I could give it a try.

Reply:
http://avivator.gehlenborglab.org/?image_url=https://viv-demo.storage.googleapis.com/B1_C1.ome.tiff should work or for a smaller stack: http://avivator.gehlenborglab.org/?image_url=https://viv-demo.storage.googleapis.com/antigen_exprs.ome.tiff

Reply:
I checked one of the sample images. The z-slice transition is already as good as it can get. That isn't really surprising because there aren't any coarse pixels that marred the transition with an image tile pyramid. There would be no benefit in putting a second tile layer on top.

Reply:
I agree, I noticed this as well.